src/FastRouteRouter.php

```diff
@@ -326,6 +381,80 @@ private function injectRoute(Route $route)
         }
 
         $this->router->addRoute($methods, $route->getPath(), $route->getPath());
-        $this->routes[$route->getName()] = $route;
+    }
+
+    /**
+     * Get the dispatch data either from cache or freshly generated by the
+     * FastRoute data generator. If caching is enabled store the freshly
+     * generated data to file.
+     *
+     * @return array
+     */
+    private function getDispatchData()
+    {
+        if ($this->hasCache) {
+            return $this->dispatchData;
+        }
+
+        $dispatchData = $this->router->getData();
+
+        if ($this->cacheEnabled) {
+            $this->cacheDispatchData($dispatchData);
+        }
+
+        return $dispatchData;
+    }
+
+    /**
+     * Load dispatchData from cache
+     *
+     * @return void
+     * @throws Exception\RuntimeException If the cache file contains invalid data
+     */
+    private function loadDispatchData()
+    {
+        if (! is_file($this->cacheFile)) {
+            return [];
+        }
+
+        $dispatchData = require $this->cacheFile;
+        if (! is_array($dispatchData)) {
+            throw new Exception\RuntimeException(
+                'Invalid cache file "' . $this->cacheFile . '". '
+                . 'The cache file MUST return an array.'
+            );
+        }
+
+        $this->hasCache = true;
+        $this->dispatchData = $dispatchData;
+    }
+
+    /**
+     * Save dispatchData to cache
+     *
+     * @param array $dispatchData
+     * @return int|false bytes written to file or false if error
+     * @throws Exception\RuntimeException
+     */
+    private function cacheDispatchData(array $dispatchData)
+    {
+        $cacheDir = dirname($this->cacheFile);
+
+        if (! is_dir($cacheDir)) {
+            throw new Exception\RuntimeException(
+                'The cache directory does not exists'
+            );
+        }
+
+        if (! is_writable($cacheDir)) {
+            throw new Exception\RuntimeException(
```

Comment:
@Ocramius 
Hello, do you mean a simple local namespace exception like https://github.com/zendframework/zend-expressive-router/blob/master/src/Exception/RuntimeException.php for instance or a properly-named execption like InvalidCacheDirectoryException ?

kind regards